test/language/expressions/optional-chaining/new-target-optional-call.js

```diff
@@ -0,0 +1,22 @@
+// Copyright 2019 Google, LLC.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: prod-OptionalExpression
+description: >
+  optional call invoked on new.target should be equivalent to call
+info: |
+  OptionalExpression
+    MemberExpression OptionalChain
+      NewTarget OptionalChain
+features: [optional-chaining]
+---*/
+
+let called = false;
+function Base() {
+  called = true;
+}
+function Foo(blerg) {
+  new.target?.();
+}
+Reflect.construct(Foo, [], Base);
+assert.sameValue(called, true);
```

Comment:
We should also check that the `this` context is correct?

Reply:
@jridgewell which `this` context:

```js
function Base() {
  return this;
}
function Foo(blerg) {
  return new.target();
}
const result = Reflect.construct(Foo, [], Base);
```

☝️ the `this` context of `Base` in `v8` appears to be the `global` object in this example, i.e.,

```
result istanceof Base // is false.
```

```js
function Foo(blerg) {
  return new.target?.();
}
```

behaves similarly.

Reply:
It should be the global in sloppy mode and `undefined` in strict.

Reply:
```js
let called = false;
function Base() {
  called = true;
  return this;
}
function Foo(blerg) {
  return new.target?.();
}
assert(Reflect.construct(Foo, [], Base) === this);
assert.sameValue(called, true);
```

Reply:
Yup. Then for `super.method?.()`, too (which has a much easier `this` context to understand).

Reply:
just adding a quick tip: `var t = (function() { return this; })();` so you reuse the default this value for sloppy and strict mode.

Reply:
@leobalter oh, If I understand, we already run tests both in both `sloppy` and `strict` mode, so I would be able to simply have one test:

```js
const t = (function() { return this; })();
assert(Reflect.construct(Foo, [], Base) === t);
```

Reply:
assuming you want the global object, yes.

For consistency, you can also load a harness file, adding to the yaml meta:

```yaml
includes: [fnGlobalObject.js]
---*/
```

so in your js you can use: `const global = fnGlobalObject();`